### PR TITLE
Add class with A, B mixin notation

### PR DESCRIPTION
**Pull request by STRd6**
Added the usage half of mixins.

```
      class A with B,C
        x = 3
      ---
      class A extends C(B(Object)) {
        x = 3
      }
```

### CODE REVIEW

File: source/parser.hera [hunk 1]
```diff
@@ -6108,6 +6128,10 @@ While
   "while" NonIdContinue ->
     return { $loc, token: $1 }
 
+With
+  "with" NonIdContinue ->
+    return { $loc, token: $1 }
```
Comment (edemaine) [inline source/parser.hera:6133]: This might be usable in `import...with` syntax too.

File: source/parser/types.civet [hunk 2]
```diff
@@ -675,3 +675,8 @@ export type VoidType = ASTLeaf & type: "VoidType"
 export type TypeLiteralNode = ASTLeaf | VoidType
 
 export type ThisAssignments = [string, ASTRef][]
+
+export type WithClause =
+  type: "WithClause"
+  children: Children
+  targets: [WSNode, ExpressionNode][]
```
Comment (edemaine) [inline source/parser/types.civet:682]: I think this needs to be added to something... `OtherNode`?

File: source/parser/util.civet [hunk 3]
```diff
@@ -229,9 +229,18 @@ function insertTrimmingSpace(target: ASTNode, c: string): ASTNode
...
+    target.replace(/^ ?/, c)
   else
     target
 
+/**
+ * Trims the first single space from the spacing array or node's children if present
+ * maintains $loc for source maps
+ */
+function trimFirstSpace(target: ASTNode): ASTNode
+  insertTrimmingSpace target, ""
```
Comment (edemaine) [inline source/parser/util.civet:242]: We use this a zillion times. I wonder if just giving `insertTrimmingSpace` an optional second argument (defaulting to empty string) would be better? Possibly also renaming it to `trimFirstSpace`...

**Review by edemaine — APPROVED** [review]
Off to a good start!

File: source/parser/types.civet (hunk 2 above)
Comment (STRd6) [reply source/parser/types.civet:682]: Currently it's consumed in the parse phase so doesn't end up in the final tree.

File: source/parser/util.civet (hunk 3 above)
Comment (STRd6) [reply source/parser/util.civet:242]: Maybe `trimFirstSpaceAndInsert` and keep the `trimFirstSpace` alias.

Comment (edemaine) [reply source/parser/util.civet:242]: `replaceFirstSpace`?

**Pull request merged**